lib/crypto.js

```diff
@@ -305,10 +305,22 @@ Sign.prototype.sign = function sign(options, encoding) {
   var passphrase = options.passphrase || null;
 
   // Options specific to RSA
-  var rsaPadding = options.padding || constants.RSA_PKCS1_PADDING;
-  var pssSaltLength = constants.RSA_PSS_SALTLEN_MAX_SIGN;
-  if (typeof options.saltLength === 'number') {
-    pssSaltLength = options.saltLength;
+  var rsaPadding = constants.RSA_PKCS1_PADDING;
+  if (options.hasOwnProperty('padding')) {
```

Comment:
Yes, that was my first consideration, but @bnoordhuis wrote:

> Can you add test cases for obviously invalid inputs (null, undefined, NaN, "boom", an object, etc.)?

If we consider `padding: undefined` invalid, then checking `padding !== undefined` is not enough. If Ben doesn't see a problem here, I will change it to `!== undefined`. Or maybe `'padding' in options` instead of `options.hasOwnProperty('padding')`.